test/test_14_file_processing.py

```diff
@@ -0,0 +1,57 @@
+class TestFileProcessing:
+    def test_thermal_video(self, helper, file_processing):
+        user = helper.admin_user()
+
+        # Ensure there's a recording to work with (the file processing
+        # API may return a different one though).
+        helper.given_a_recording(self)
+
+        # Get a recording to process.
+        recording = file_processing.get("thermalRaw", "getMetadata")
+        assert recording["processingState"] == "getMetadata"
+
+        # Move job to next stage.
+        file_processing.put(
+            recording,
+            success=True,
+            complete=False,
+            updates={"additionalMetadata": {"one": "foo"}},
+        )
+        check_recording(
+            user,
+            recording["id"],
+            processingState="toMp4",
+            additionalMetadata={"one": "foo"},
+        )
+
+        # Now finalise processing.
+        file_processing.put(
+            recording,
+            success=True,
+            complete=True,
+            updates={
+                "additionalMetadata": {"two": "bar"},
+                "fileMimeType": "application/cheese",
+            },
+            new_object_key=recording["rawFileKey"],
+        )
+        check_recording(
+            user,
+            recording["id"],
+            processingState="FINISHED",
+            fileMimeType="application/cheese",
+            additionalMetadata={"one": "foo", "two": "bar"},  # note: merged
```

Comment:
Yay new tests! 

Might be better as two tests though.  
1.   That checks the recording state changes and complete works.  
2.  That checks additionalMetadata merges. 

Also may want to check that you can override meta data too.

Reply:
ok, I'll split it up